village/src/components/SmurfForm.js

```diff
@@ -4,31 +4,32 @@ class SmurfForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      name: '',
-      age: '',
-      height: ''
-    };
-  }
-
-  addSmurf = event => {
-    event.preventDefault();
-    // add code to create the smurf using the api
-
-    this.setState({
-      name: '',
-      age: '',
-      height: ''
-    });
-  }
+        name: '',
+        age: '',
+        height: ''
+      }
+}
 
   handleInputChange = e => {
-    this.setState({ [e.target.name]: e.target.value });
+    this.setState({ 
+      ...this.state,
+      [e.target.name]: e.target.value
+    });
   };
 
+
+  addSmurf = e => {
+    e.preventDefault();
+    console.log("clicked")
+    this.props.addSmurf(this.state)
```

Comment:
add `this.props.history.push('/')` to route to the smurf list after adding a smurf